internal/providercache/package_install.go

```diff
@@ -136,7 +138,48 @@ func installFromLocalArchive(ctx context.Context, meta getproviders.PackageMeta,
 	// match the allowed hashes and so our caller should catch that after
 	// we return if so.
 
-	err := unzip.Decompress(targetDir, filename, true, 0000)
+	err := os.MkdirAll(targetDir, 0777)
+	if err != nil {
+		return authResult, fmt.Errorf("failed to create new directory: %w", err)
+	}
+
+	stagingDir, err := os.MkdirTemp(path.Dir(targetDir), ".temp")
```

Comment:
Presumably we could get rid of the need to refactor the `SearchLocalDirectory` function if we wrote the files into a directory created by `os.TempDir()` instead of inside our cache? Any reason not to do that?

Reply:
A reason not to do that is that the temporary filesystem could be on a different device; therefore, `os.Rename` could fail. https://github.com/golang/go/issues/13766